grpc-gcp/src/main/java/com/google/cloud/grpc/GcpManagedChannelOptions.java

```diff
@@ -276,6 +318,72 @@ public Builder setMinSize(int minSize) {
         return this;
       }
 
+      /**
+       * Sets the initial channel pool size. This is the number of channels that the pool will start
+       * with. If it is less than {@link #setMinSize(int)} it is ignored.
+       *
+       * @param initSize number of channels to start the pool with.
+       * @return
+       */
+      public Builder setInitSize(int initSize) {
+        Preconditions.checkArgument(
+            initSize >= 0, "Channel pool initial size must be 0 or positive.");
+        this.initSize = initSize;
+        return this;
+      }
+
+      /**
+       * Enables dynamic scaling functionality.
+       *
+       * <p>When the average number of concurrent calls per channel reaches <code>maxRpcPerChannel
+       * </code> the pool will create and add a new channel unless already at max size.
+       *
+       * <p>Every <code>scaleDownInterval</code> a check for downscaling is performed. Based on the
+       * maximum total concurrent calls observed since the last check, the desired number of
```

Comment:
Why the total concurrent calls is observed since the last check, instead of the current number?

Reply:
The total here means across all channels. And it is the _maximum_ total concurrent calls, i.e. it is _not_ total calls since last check but the largest number of concurrent calls we saw at a single moment since the last check.

And we look at that instead of current number of concurrent calls to protect from situation when at the moment of a check the number of concurrent calls is low but between the checks it is high.

Does this make sense?

Reply:
I see, thanks for the explanation!